contracts/sdk/IpcContract.sol

```diff
@@ -0,0 +1,115 @@
+// SPDX-License-Identifier: MIT OR Apache-2.0
+pragma solidity 0.8.19;
+
+import { IpcEnvelope, ResultMsg, CallMsg, IpcMsgKind } from "../src/structs/CrossNet.sol";
+import { IPCAddress } from "../src/structs/Subnet.sol";
+import { EMPTY_BYTES } from "../src/constants/Constants.sol";
+import { IGateway } from "../src/interfaces/IGateway.sol";
+import { ReentrancyGuard } from "openzeppelin-contracts/security/ReentrancyGuard.sol";
+import { Ownable } from "openzeppelin-contracts/access/Ownable.sol";
+import { CrossMsgHelper } from "../src/lib/CrossMsgHelper.sol";
+
+// Interface that needs to be implemented by IPC-aware contracts.
+//
+// TODO: extract to a shared module, so that both the IPC contracts and the SDK can depend on this.
+//  (IPC contracts reaching into the SDK is wrong).
+interface IpcHandler {
+    error CallerIsNotGateway();
+    error UnsupportedMsgKind();
+    error UnrecognizedResult();
+
+    /// @notice Entrypoint for handling xnet messages in IPC-aware contracts.
+    function handleIpcMessage(IpcEnvelope calldata envelope) external payable returns (bytes memory ret);
+}
+
+abstract contract IpcExchange is IpcHandler, Ownable {
```

Comment:
@raulk should this file be renamed to `IpcExchange.sol` or `IpcHandler` to avoid introducing more names?